### PR TITLE
MultiTerminal tests

### PR DESCRIPTION
**Pull request by simplemachine92**

### CODE REVIEW

File: forge_tests/TestLaunchProject.sol [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        // Package up terminal config.
+        JBTerminalConfig[] memory _terminalConfigurations = new JBTerminalConfig[](1);
+        JBAccountingContextConfig[] memory _accountingContexts = new JBAccountingContextConfig[](1);
+        _accountingContexts[0] =
+            JBAccountingContextConfig({token: JBTokens.ETH, standard: JBTokenStandards.NATIVE});
+        _terminalConfigurations[0] =
+            JBTerminalConfig({terminal: _terminal, accountingContextConfigs: _accountingContexts});
+
+        // expectRevert on the next call if weight overflowing
+        if (_weight > type(uint88).max) {
```
Comment (drgorillamd) [inline forge_tests/TestLaunchProject.sol:113]: Fuzzing-wise, this creates a huge skew -> wdyt of rather use bound here (_weight ∈ [0; uint88.max -1] ) and have a small other test for > 2^88)?
-> rn, a lots of the fuzzed weight are falling in [2^88; 2^256], as forge fuzzer isn't random (or pure gaussian), it's (iirc) just a big kurtosis, here, we "loose" all the runs from both the mean and the right fat tail (most of the info for us is in the < 2^88 interval)

Comment (drgorillamd) [reply forge_tests/TestLaunchProject.sol:113]: If you see this pattern somewhere else, this should be part of overall refactoring (same as using vm.assume to select interval, for the same reason as vm.assume just drop the run while bound recenters it)

File: forge_tests/TestOperatorStore.sol [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    function testFailMostBasicAccess() public {
+        // Package up cycle config.
+        JBFundingCycleConfig[] memory _cycleConfig = new JBFundingCycleConfig[](1);
+        _cycleConfig[0].mustStartAtOrAfter = 0;
+        _cycleConfig[0].data = _data;
+        _cycleConfig[0].metadata = _metadata;
+        _cycleConfig[0].groupedSplits = new JBGroupedSplits[](0);
+        _cycleConfig[0].fundAccessConstraints = new JBFundAccessConstraints[](0);
+        
+        vm.prank(makeAddr("8008"));
```
Comment (drgorillamd) [inline forge_tests/TestOperatorStore.sol:99]: avoid funny values -> global `projectZeroOwner` ?

File: forge_tests/TestOperatorStore.sol [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        _controller.reconfigureFundingCyclesOf(_projectOne, _cycleConfig, "");
```
Comment (drgorillamd) [inline forge_tests/TestOperatorStore.sol:100]: no assertion?

File: forge_tests/TestOperatorStore.sol [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                domain: _projectOne,
+                permissionIndexes: permIndexes
+                });
+
+        // Set em.
+        vm.prank(_projectOwner);
+        _opStore.setOperatorOf(_projectOwner, opData[0]);
+
+        // verify
+        bool _check = _opStore.hasPermission(address(0), _projectOwner, _projectOne, permIndexes[i]);
```
Comment (drgorillamd) [inline forge_tests/TestOperatorStore.sol:146]: nit> in another loop

Maybe it isn't super consistent in our current codebase (pleading guilty as charged), but we should (try to) use a formalised template, something like Setup/Test/Check comments to have clearer structure accross tests, as well as include natspecs 

```
// Setup: Create array to pass to foo() to do magic stuff
```
creating some arrays and stuff, calling some other fn (if integration) or setting mocks (if units)

```
// Test: Caling foo()
```
the tx we test now (or multiple tx)

```
// Check: Is result greater than Namibia GDP?
```
whatever assert/loop assert/etc (we group the assert in a loop even if test are calling in a loop)

File: forge_tests/TestLaunchProject.sol (hunk 1 above)
Comment (simplemachine92) [reply forge_tests/TestLaunchProject.sol:134]: Looks good but I think should be (_weight ∈ [0; uint88.max])

File: forge_tests/TestOperatorStore.sol (hunk 3 above)
Comment (simplemachine92) [reply forge_tests/TestOperatorStore.sol:100]: Felt implicit to me as if reconfigure fails, as will the test, but ofc could fail for various reasons (if misconfigured).

What is the best way for readability though? First thought is assertEq(configured == block.timestamp).

File: forge_tests/TestOperatorStore.sol [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            permIndexes[i] = i;
+
+            opData[0] = JBOperatorData({
+                operator: address(0),
+                domain: _projectOne,
+                permissionIndexes: permIndexes
+                });
+
+        // Set em.
+        vm.prank(_projectOwner);
```
[the remark this answers is not in the document]
Comment (simplemachine92) [reply forge_tests/TestOperatorStore.sol:120]: Implicit and expected to fail, but what would you prefer for readability?

**Pull request merged**